sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -106,12 +106,20 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
 
       // Filter the plan by applying left semi and left anti joins.
       withSubquery.foldLeft(newFilter) {
-        case (p, Exists(sub, conditions, _)) =>
-          val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
-          buildJoin(outerPlan, sub, LeftSemi, joinCond)
-        case (p, Not(Exists(sub, conditions, _))) =>
-          val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
-          buildJoin(outerPlan, sub, LeftAnti, joinCond)
+        case (p, exists @ Exists(sub, conditions, _)) =>
+          if (SubqueryExpression.hasCorrelatedSubquery(exists)) {
+            val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
+            buildJoin(outerPlan, sub, LeftSemi, joinCond)
+          } else {
+            Filter(exists, newFilter)
+          }
+        case (p, Not(exists @ Exists(sub, conditions, _))) =>
+          if (SubqueryExpression.hasCorrelatedSubquery(exists)) {
+            val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
+            buildJoin(outerPlan, sub, LeftAnti, joinCond)
+          } else {
+            Filter(Not(exists), newFilter)
+          }
```

Comment:
@cloud-fan @dilipbiswal 
Change here to support  non-correct exists subquery run like mentioned in this comment https://github.com/apache/spark/pull/26437/files#r344203937

Reply:
@AngersZhuuuu I discussed this with Wenchen. Do you think we can safely inject a "LIMIT 1" into our subplan to expedite its execution ? Pl. lets us know what you think ?

Reply:
> @AngersZhuuuu I discussed this with Wenchen briefly. Do you think we can safely inject a "LIMIT 1" into our subplan to expedite its execution ? Pl. lets us know what you think ?

I am also thinking about reduce the execution cost of this sub query. 
`LIMIT 1` is ok . 
My direction is making this execution  like Spark Thrift Server's incremental collect.
Only execute one partition.

Discuss these two ways safety and cost?